cognition_objects/pipeline_log.py

```diff
@@ -220,3 +221,63 @@ def update_to_new_diff_structure(
 
     if with_commit:
         general.commit()
+
+
+def get_error_and_time_elapsed_by_conversation_id(
+    project_id: str,
+    conversation_id: str,
+) -> List[CognitionPipelineLogs]:
+    grouped_message_logs = (
+        session.query(
+            CognitionMessage.id,
+            func.sum(CognitionPipelineLogs.time_elapsed).label("time_logs_elapsed"),
+            func.bool_or(CognitionPipelineLogs.has_error).label("logs_have_error"),
+        )
+        .join(CognitionMessage)
+        .join(CognitionConversation)
+        .filter(
+            CognitionPipelineLogs.project_id == project_id,
+            CognitionPipelineLogs.message_id == CognitionMessage.id,
+            CognitionMessage.conversation_id == conversation_id,
+        )
+        .group_by(CognitionMessage.id)
+        .all()
+    )
+    grouped_logs_dict = {}
+    for message in grouped_message_logs:
+        grouped_logs_dict[message.id] = {
+            "time_logs_elapsed": message.time_logs_elapsed,
+            "logs_have_error": message.logs_have_error,
+        }
```

Comment:
join should include a join criteria. With the added dict requirement i would recommend an sql query

```
SELECT jsonb_object_agg(id, jsonb_build_object('has_error',has_error, 'time_elapsed',time_elapsed))
FROM (
	SELECT m.id, SUM(pl.has_error::INT) > 0 has_error,SUM(pl.time_elapsed) time_elapsed
	FROM cognition.message m
	INNER JOIN cognition.pipeline_logs pl
		ON m.project_id = pl.project_id AND m.id = pl.message_id
	WHERE m.project_Id = '72c5721d-6cd4-4652-97b7-3aeca0600e07'
	-- AND m.conversation_id = '9fe61702-120e-4035-8b4b-e037ffb8812c'
	-- AND m.message_id = 'd5d547cc-e5ef-41e4-87f8-a53e578e6bcc'
	GROUP BY m.id 
)x
```